src/snowflake/cli/plugins/snowpark/package_utils.py

```diff
@@ -72,7 +73,11 @@ def get_snowflake_packages() -> List[str]:
         with requirements_file.open(
             "r", read_file_limit_mb=DEFAULT_SIZE_LIMIT_MB, encoding="utf-8"
         ) as f:
-            return [req for line in f if (req := line.split("#")[0].strip())]
+            return [
+                Requirement.parse(req).to_name_and_version()
+                for line in f
+                if (req := line.split("#")[0].strip())
```

Comment:
Duplicated code fragment, consider introducing a method for cleaning comments.

Reply:
After refactor it is used only in `snowflake.cli.plugins.snowpark.package_utils.parse_requirements`